### PR TITLE
Bump nmstate to 1.0

### PR DESCRIPTION
**Pull request by qinqon**
Signed-off-by: Quique Llorente <ellorent@redhat.com>

**Is this a BUG FIX or a FEATURE ?**:

/kind enhancement

**What this PR does / why we need it**:
Let's move nmstate to 1.0 at master.

**Special notes for your reviewer**:

**Release note**:

```release-note
Bump nmstate to 1.0
```

### CODE REVIEW

**Review by phoracek — COMMENTED** [review]
Would you please use this to ditch the compatibility layer in our tests? i.e. drop "slaves".

File: Makefile [hunk 1]
```diff
@@ -4,11 +4,11 @@ IMAGE_REGISTRY ?= quay.io
...
 NAMESPACE ?= nmstate
 
-NMSTATE_CURRENT_COPR_REPO=nmstate-0.3
-NM_CURRENT_COPR_REPO=NetworkManager-1.26
+NMSTATE_CURRENT_COPR_REPO=nmstate-1.0
+NM_CURRENT_COPR_REPO=NetworkManager-1.30
 
 NMSTATE_FUTURE_COPR_REPO=nmstate
-NM_FUTURE_COPR_REPO=NetworkManager-1.26
+NM_FUTURE_COPR_REPO=NetworkManager-1.30
```
Comment (phoracek) [inline Makefile:11]: We can make this useful by using  nmstate/nmstate-git. nmstate/nmstate is the same as 1.0

**Comment by phoracek** [comment]
Still applies: https://github.com/nmstate/kubernetes-nmstate/pull/745#pullrequestreview-654450816

File: Makefile (hunk 1 above)
Comment (qinqon) [reply Makefile]: Done

**Comment by oshoval** [comment]
> **Is this a BUG FIX or a FEATURE ?**:
> 
> > Uncomment only one, leave it on its own line:
> > /kind bug
> > /kind enhancement

Please remove this or select one

File: Makefile [hunk 2]
```diff
@@ -4,11 +4,11 @@ IMAGE_REGISTRY ?= quay.io
...
 
-NMSTATE_CURRENT_COPR_REPO=nmstate-0.3
-NM_CURRENT_COPR_REPO=NetworkManager-1.26
+NMSTATE_CURRENT_COPR_REPO=nmstate-1.0
+NM_CURRENT_COPR_REPO=NetworkManager-1.30
 
-NMSTATE_FUTURE_COPR_REPO=nmstate
-NM_FUTURE_COPR_REPO=NetworkManager-1.26
+NMSTATE_FUTURE_COPR_REPO=nmstate-git
+NM_FUTURE_COPR_REPO=NetworkManager-main
```
Comment (EdDev) [inline Makefile:11]: In the nmstate case, I think this tracks master, including non-released versions. Are you sure you want to depend on non-released code?

Comment (qinqon) [reply Makefile:11]: Right now the "nmstate" copr repo points is the same at "nmstate-1.0" so our only way to test on future changes is with "nmstate-git", let's try this, if we see it breaks too often we go back "nmstate".

**Comment by qinqon** [comment]
Looks like nmstate-1.0 is slower and prow jobs are timing out.

**Comment by phoracek** [comment]
I'm pretty sure we have been suffering timeouts even before this bump.

**Comment by qinqon** [comment]
On the Spec is failing 
```
Failure in Spec Setup (BeforeEach) [518.335 seconds]
```

So job duration is increasing by 8 minutes or so.

File: docs/examples/bond-vlan.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ spec:
         mode: balance-rr
         options:
           miimon: '140'
-        slaves:
+        ports:
```
Comment (rhrazdil) [inline docs/examples/bond-vlan.yaml:18]: `git grep -n slaves` shows a few more references in our documentation, might be worth changing those at the same time 

```
docs/user-guide/102-configuration.md:169:        slaves:
docs/user-guide/102-configuration.md:214:        slaves:
docs/user-guide/103-troubleshooting.md:75:      autoconnect_slaves: <enum NM_SETTING_CONNECTION_AUTOCONNECT_SLAVES_YES of type NM.SettingConnectionAutoconnectSlaves>
docs/user-guide/bond0-eth1-eth2_up.yaml:17:        slaves:
```

**Review by rhrazdil — CHANGES_REQUESTED** [review]

File: docs/examples/bond-vlan.yaml (hunk 3 above)
Comment (qinqon) [reply docs/examples/bond-vlan.yaml:18]: Done

File: docs/user-guide/103-troubleshooting.md [hunk 4]
```diff
@@ -72,7 +72,7 @@ status:
       uuid: 70b556e5-e2da-4e46-a98f-531bc73d0bf5
       type: 802-3-ethernet
       autoconnect: True
-      autoconnect_slaves: <enum NM_SETTING_CONNECTION_AUTOCONNECT_SLAVES_YES of type NM.SettingConnectionAutoconnectSlaves>
+      autoconnect_ports: <enum NM_SETTING_CONNECTION_AUTOCONNECT_SLAVES_YES of type NM.SettingConnectionAutoconnectSlaves>
```
Comment (phoracek) [inline docs/user-guide/103-troubleshooting.md:75]: Is this really what you see in the traceback? We must be polite while truthful.

**Review by rhrazdil — APPROVED** [review]
Thank you!
/approve

**Comment by rhrazdil** [comment]
/hold
Missed Peter's comment

File: docs/user-guide/103-troubleshooting.md (hunk 4 above)
Comment (qinqon) [reply docs/user-guide/103-troubleshooting.md:75]: Right is still autoconnect_slaves

**Review by phoracek — COMMENTED** [review]
/lgtm

**Comment by qinqon** [comment]
/retest

operator test still flaky

**Pull request merged**